### PR TITLE
レビュー用:コメント追加

### PR DESCRIPTION
**Pull request by acha0000**

### CODE REVIEW

File: 1-3_transfer_learning.ipynb [hunk 1]
```diff
@@ -132,31 +157,39 @@
...
+        "\n",
         "std = (0.229, 0.224, 0.225)#標準偏差を設定\n",
         "\n",
         "transform = ImageTransform(size, mean, std)#ImageTransformをつかって↑のさっき作った処理をさせる\n",
         "#RandomResizedCropランダムに切り取って拡大縮小 RandomHorizontalFlip()ランダムに左右反転 ToTensor()Pytorchの見やすい形（テンソル）させる Normalize(mean,std)ここで標準化させる\n",
+        "\n",
         "img_transformed = transform(img, phase=\"train\")  # torch.Size([3, 224, 224])テンソル形式にして、色のチャンネル数×高さ×幅にする\n",
         "\n",
         "# (色、高さ、幅)を (高さ、幅、色)に変換し、0-1に値を制限して表示\n",
         "img_transformed = img_transformed.numpy().transpose((1, 2, 0))#numpyでテンソルをNumpyの配列に変える（意味わからんｗｗ）\n",
```
Comment (yuki-mtmr) [inline 1-3_transfer_learning.ipynb:181]: ここは以下の理由でNumpy配列にする。(画像を扱うときの形状の順番はpytorchは違う

理由①: Matplotlib で表示するため
torch.Tensor のままでは matplotlib.pyplot.imshow() で正しく表示できない。
imshow() は NumPy の ndarray を受け付けるため、 PyTorch の Tensor を NumPy に変換する必要がある。
理由②: Tensor のデータ形式（チャンネル順）が異なるため
PyTorch の Tensor の形状は (色チャネル, 高さ, 幅) (C, H, W)
matplotlib は (高さ, 幅, 色チャネル) (H, W, C) を想定しているため、transpose((1, 2, 0)) でチャンネルの順番を入れ替える。
理由③: clip(0, 1) でデータ範囲を制限するため
PyTorch の Tensor の範囲は [-1, 1] になることがある（Normalize(mean, std) の影響）。

File: 1-3_transfer_learning.ipynb [hunk 2]
```diff
@@ -2,48 +2,64 @@
...
         "id": "q2NVkjTuDJbv"
       },
       "outputs": [],
       "source": [
         "# 乱数のシードを設定毎回違う結果を出すため\n",
         "torch.manual_seed(1234)#これはPytorchの乱数を固定\n",
+        "\n",
         "np.random.seed(1234)#Numpyの乱数を固定する\n",
+        "\n",
         "random.seed(1234)#PythonのRandomモジュールの乱数を固定する）データのランダム選択などに影響"
```
Comment (yuki-mtmr) [inline 1-3_transfer_learning.ipynb:57]: ここで乱数シードで同じ値を渡すのは毎回同じ結果を出すため。
シードが同じなら必ず同じランダムの結果が出る。そうしないと実行の度に最終結果も変わるからちゃんと検証できない。

File: 1-3_transfer_learning.ipynb [hunk 3]
```diff
@@ -2,48 +2,64 @@
...
         "import json #辞書みたいなデータを読み書きするため\n",
+        "\n",
         "from PIL import Image #ここから↓は画像系　画像を開いたり、サイズを変えたりできる\n",
+        "\n",
         "from tqdm import tqdm #進行状況をバーで出してくれる\n",
+        "\n",
         "import matplotlib.pyplot as plt #画像やデータをグラフで表示\n",
+        "\n",
         "%matplotlib inline\n",
         "#↑Jupyter Notebookでplt.show()を使う必要があったが使わずに画像を表示できる\n",
```
Comment (yuki-mtmr) [inline 1-3_transfer_learning.ipynb:29]: 以前は matplotlib のグラフを自動表示するために %matplotlib inline が必要だったが、現在の Jupyter Notebook ではデフォルトで有効になっているため、基本的に不要。
ただし、古い環境との互換性や「Matplotlib を使用している」と明示する目的で使うことは有用。
インタラクティブな操作をしたい場合は %matplotlib notebook や %matplotlib widget を使用。

**Review by yuki-mtmr — COMMENTED** [review]
気になったとこだけコメント残したよ。

あとそんな大したことじゃないけどL814の↓はPythonじゃなくて"PyTorchでは"かな
Pythonではデフォルトで勾配が蓄積するからリセットする必要があるため初期化

File: 1-3_transfer_learning.ipynb (hunk 1 above)
Comment (acha0000) [reply 1-3_transfer_learning.ipynb:181]: ここでつかうPytorchは画像を配列を使って表すために使ってるってことの僕の解釈なんですがあってますか？

Comment (yuki-mtmr) [reply 1-3_transfer_learning.ipynb:181]: それは合ってるよ。
Pytorchを使って画像データをディープラーニングで使える配列(テンソル)の形に変えてる。ただ、pytorchは画像の形状を扱う順番がmatplotlibで扱える形状とちがうから画像を表示するときには形状を一回直さないと画像表示できない。

- PyTorch: 画像を (C, H, W) で扱う
- Matplotlib: 画像を (H, W, C) で扱う
-  表示するには transpose((1,2,0)) で形を戻す必要がある！

C(色),H(高さ),W(幅)
